posthog/temporal/weekly_digest/activities.py

```diff
@@ -546,9 +551,12 @@ async def send_weekly_digest_batch(input: SendWeeklyDigestBatchInput) -> None:
                     continue
                 finally:
                     if not input.dry_run and partial:
-                        ph_client.flush()
                         messaging_record.sent_at = timezone.now()
-                        await messaging_record.asave()
+                        messaging_record_batch.append(messaging_record)
+
+                    if len(messaging_record_batch) >= RECORD_BATCH_SIZE:
+                        await MessagingRecord.objects.abulk_update(messaging_record_batch, ["sent_at"])
+                        messaging_record_batch = []
```

Comment:
do you risk having some dangling unsaved entries in messaging_record_batch after the loop?

Reply:
I did and it caused a test to fail so just fixed it 👍